File: backend/src/main/java/by/fpmibsu/findafriend/controller/commands/adverts/UpdateAdvertHandler.java
package by.fpmibsu.findafriend.controller.commands.adverts;

import by.fpmibsu.findafriend.application.Application;
import by.fpmibsu.findafriend.application.mediatr.RequestHandler;
import by.fpmibsu.findafriend.dataaccesslayer.advert.AdvertDao;

public class UpdateAdvertHandler extends RequestHandler<Boolean, UpdateAdvertCommand> {
    private final AdvertDao advertDao;

    public UpdateAdvertHandler(AdvertDao advertDao) {
        this.advertDao = advertDao;
    }

    @Override
    public Boolean handle(UpdateAdvertCommand request) {
        var advert = advertDao.getEntityById(request.advertId);
        if (advert == null) {
            return false;
        }
        if (request.advertType != null) {
            advert.setAdvertType(request.advertType);
        }
        if (request.title != null) {
            advert.setTitle(request.title);
        }
        if (request.description != null) {
            advert.setDescription(request.description);
        }
        if (request.creationDate != null) {
            advert.setCreationDate(request.creationDate);
        }
        if (request.place != null) {
            advert.setPlace(request.place);
        }

        advertDao.update(advert);
        return true;
    }
}

File: BackendNET/Backend/Commands/Adverts/UpdateAdvert/UpdateAdvertHandler.cs
using Backend.Repository;
using MediatR;

namespace Backend.Commands.Adverts.UpdateAdvert;

public class UpdateAdvertHandler : IRequestHandler<UpdateAdvertCommand, bool>
{
    private readonly IAdvertRepository _repository;

    public UpdateAdvertHandler(IAdvertRepository repository)
    {
        _repository = repository;
    }

    public async Task<bool> Handle(UpdateAdvertCommand request, CancellationToken cancellationToken)
    {
        var instance = await _repository.GetByIdAsync(request.AdvertId, cancellationToken);
        if (instance is null)
        {
            return false;
        }

        instance.Type = request.AdvertType ?? instance.Type;
        instance.PlaceId = request.PlaceId ?? instance.PlaceId;
        instance.Title = request.Title ?? instance.Title;
        instance.Description = request.Description ?? instance.Description;
        await _repository.UpdateAsync(instance, cancellationToken);
        return true;
    }
}